librad/src/git/types/remote.rs

```diff
@@ -100,6 +100,11 @@ impl<Url> Remote<Url> {
         }
     }
 
+    pub fn with_refspec(mut self, refspec: Box<dyn AsRefspec>) -> Self {
```

Comment:
I feel like we have all permutations of `with_x(mut self) -> Self`, `set_x(&mut self) -> &mut Self`, `with_x(self) -> Self`, `x(&mut self) -> &mut Self`, `set_x(&mut self) -> ()`, etc. somewhere in the code. I'll book a `C_IDIOTMATIC` workshop with the styleguide polizei for next year, ok?

Reply:
Yaw, we have them here https://github.com/radicle-dev/radicle-link/blob/next/librad/src/git/types/reference.rs#L127-L140.

We need polymorphic ownership lenses :pray:

Reply:
Who needs polymorphic anything if there are CODING CONVENTIONS?!